backtester/eventhandlers/strategies/ftxcashandcarry/ftxcashandcarry.go

```diff
@@ -53,63 +56,125 @@ func (s *Strategy) OnSimultaneousSignals(d []data.Handler, f funding.IFundingTra
 		return nil, errNoSignals
 	}
 	if f == nil {
-		return nil, fmt.Errorf("%w missing funding transferred", common.ErrNilArguments)
+		return nil, fmt.Errorf("%w missing funding transferred", gctcommon.ErrNilPointer)
 	}
 	if p == nil {
-		return nil, fmt.Errorf("%w missing portfolio handler", common.ErrNilArguments)
+		return nil, fmt.Errorf("%w missing portfolio handler", gctcommon.ErrNilPointer)
 	}
 	var response []signal.Event
 	sortedSignals, err := sortSignals(d)
 	if err != nil {
 		return nil, err
 	}
 
-	for _, v := range sortedSignals {
-		pos, err := p.GetPositions(v.futureSignal.Latest())
-		if err != nil {
-			return nil, err
-		}
-		spotSignal, err := s.GetBaseData(v.spotSignal)
-		if err != nil {
-			return nil, err
-		}
-		futuresSignal, err := s.GetBaseData(v.futureSignal)
-		if err != nil {
-			return nil, err
-		}
+	for _, b := range sortedSignals {
+		for _, v := range b {
```

Comment:
Seems like you are not using the keys for the currency items here. Might be better to return just a slice of cashCarrySignals instead?